src/librustc/hir/map/collector.rs

```diff
@@ -115,6 +115,13 @@ impl<'ast> Visitor<'ast> for NodeCollector<'ast> {
                     // If this is a tuple-like struct, register the constructor.
                     if !struct_def.is_struct() {
                         this.insert(struct_def.id(), NodeStructCtor(struct_def));
+                    } else {
+                        // This should be replaced with `visit_struct_field` once there's a Node
+                        // for struct fields. In the meantime, register the parent node to have
+                        // something to point to.
```

Comment:
I hadn't seen that PR. That's quite a refactor! I have no problem waiting for it.